src/messaging/tests/TestReliableMessageProtocol.cpp

```diff
@@ -1443,6 +1443,78 @@ void CheckLostStandaloneAck(nlTestSuite * inSuite, void * inContext)
     NL_TEST_ASSERT(inSuite, rm->TestGetCountRetransTable() == 0);
 }
 
+struct BackoffComplianceTestVector
+{
+    uint8_t sendCount;
+    System::Clock::Timestamp backoffBase;
+    System::Clock::Timestamp backoffMin;
+    System::Clock::Timestamp backoffMax;
+};
+
+struct BackoffComplianceTestVector theBackoffComplianceTestVector[] = {
+    {
+        .sendCount   = 0,
+        .backoffBase = System::Clock::Timestamp(300),
+        .backoffMin  = System::Clock::Timestamp(300),
+        .backoffMax  = System::Clock::Timestamp(375),
+    },
+    {
+        .sendCount   = 1,
+        .backoffBase = System::Clock::Timestamp(300),
+        .backoffMin  = System::Clock::Timestamp(300),
+        .backoffMax  = System::Clock::Timestamp(375),
+    },
+    {
+        .sendCount   = 2,
+        .backoffBase = System::Clock::Timestamp(300),
+        .backoffMin  = System::Clock::Timestamp(480),
+        .backoffMax  = System::Clock::Timestamp(600),
+    },
+    {
+        .sendCount   = 3,
+        .backoffBase = System::Clock::Timestamp(300),
+        .backoffMin  = System::Clock::Timestamp(768),
+        .backoffMax  = System::Clock::Timestamp(960),
+    },
+    {
+        .sendCount   = 4,
+        .backoffBase = System::Clock::Timestamp(300),
+        .backoffMin  = System::Clock::Timestamp(1229),
+        .backoffMax  = System::Clock::Timestamp(1536),
+    },
+    {
+        .sendCount   = 5,
+        .backoffBase = System::Clock::Timestamp(300),
+        .backoffMin  = System::Clock::Timestamp(1966),
+        .backoffMax  = System::Clock::Timestamp(2458),
+    },
+    {
+        .sendCount   = 6,
+        .backoffBase = System::Clock::Timestamp(300),
+        .backoffMin  = System::Clock::Timestamp(1966),
+        .backoffMax  = System::Clock::Timestamp(2458),
+    },
+};
+
+const unsigned theBackoffComplianceTestVectorLength =
+    sizeof(theBackoffComplianceTestVector) / sizeof(struct BackoffComplianceTestVector);
+
+void CheckGetBackoff(nlTestSuite * inSuite, void * inContext)
+{
+    TestContext & ctx       = *reinterpret_cast<TestContext *>(inContext);
+    ReliableMessageMgr * rm = ctx.GetExchangeManager().GetReliableMessageMgr();
+
+    for (unsigned i = 0; i < theBackoffComplianceTestVectorLength; i++)
+    {
+        struct BackoffComplianceTestVector * test = &theBackoffComplianceTestVector[i];
+        System::Clock::Timestamp backoff          = rm->GetBackoff(test->backoffBase, test->sendCount);
```

Comment:
nit: any value in looping a bit over every test vector to try different jitter values just in case? seems fast enough to compute.

Reply:
Are you suggesting to feed the test undefined values as a sort of fuzzing technique?  Two issues with that:
1) That wouldn't work because the test vector includes expected values or "truth data", which random values would not get right.
2) Undefined values are unlikely to have an evenly random spread.  A high percentage of the inputs would be simply `0`, as such, explicit fuzz tests are better in those cases.

Reply:
I was thinking to do each of the test vector test a few times because of the jitter using Rnd data, so to validate we get within min and max every time instead of only once.

Reply:
Ah ok.  I can make that change, sure.